connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetrics.java

```diff
@@ -129,16 +133,22 @@ public MetricGroup group(String groupName, String... tagKeyValues) {
      * @throws IllegalArgumentException if the group name is not valid
      */
     public MetricGroup group(String groupName, boolean includeWorkerId, String... tagKeyValues) {
-        MetricGroup group = groupsByName.get(groupName);
+        MetricGroupId groupId = groupId(groupName, includeWorkerId, tagKeyValues);
+        MetricGroup group = groupsByName.get(groupId);
         if (group == null) {
-            Map<String, String> tags = tags(includeWorkerId ? workerId : null, tagKeyValues);
-            group = new MetricGroup(groupName, tags);
-            MetricGroup previous = groupsByName.putIfAbsent(groupName, group);
+            group = new MetricGroup(groupId);
+            MetricGroup previous = groupsByName.putIfAbsent(groupId, group);
             if (previous != null) group = previous;
         }
         return group;
     }
 
+    protected MetricGroupId groupId(String groupName, boolean includeWorkerId, String... tagKeyValues) {
+        checkNameIsValid(groupName);
+        Map<String, String> tags = tags(includeWorkerId ? workerId : null, tagKeyValues);
```

Comment:
We missed it in the first round probably because those metrics only had 1 tag, but the `tags` method should use something like a `LinkedHashMap`. Without this, the tags can get jumbled. I noticed this because jconsole's hierarchy was listing the task ID above the connector name.

Reply:
Yup.